script.js

```diff
@@ -0,0 +1,186 @@
+      emptyCommit: false
+    }
+  ];
+
+  /**
+   * @param {Exclude<import('@octokit/openapi-types').components["schemas"]["content-tree"]['entries'], undefined>[number] | import('@octokit/openapi-types').components["schemas"]["content-tree"]} file
+   * @param {import('octokit-plugin-create-pull-request').createPullRequest.UpdateFunctionFile} options
+   * @returns {string | null}
+   */
+  function updateYamlFile(file, { content, encoding, exists }) {
+    if (!exists) return null;
+
+    octokit.log.info('Checking \'%s\' in \'%s\' repo', file.name, repo);
+
+    if (hasNodeVersionToRemove(content, NODE_VERSIONS_TO_REMOVE)) {
+      octokit.log.warn(
+          'The repository \'%s\' HAS a node_version %s to be removed in its GitHub Actions.\n %s',
+          repo,
+          `${NODE_VERSIONS_TO_REMOVE}`,
+          repository.html_url
+      );
+
+      // Update Node versions used in GitHub Actions
+      const yamlDocument = parseDocument(
+          Buffer.from(content, encoding).toString('utf-8')
+      );
+
+      /** @type {import('yaml').YAMLMap<string>} */
+      // @ts-expect-error Why is this `unknown`?
+      const jobs = yamlDocument.get('jobs');
+
+      for (const { value: job, key: jobName } of jobs.items) {
+        /** @type {import('yaml').YAMLSeq<number> | undefined} */
+        const nodeVersions = job
+            .get('strategy')
+            ?.get('matrix')
+            ?.get('node_version');
+
+        if (nodeVersions) {
+          yamlDocument.setIn(['jobs', jobName, 'strategy', 'matrix', 'node_version'], NODE_VERSIONS);
+        }
+      }
+
+      return prettier.format(stringify(yamlDocument), { parser: 'yaml' });
+    }
+    octokit.log.info(
+        'The repository \'%s\' does not have any usage of node_version %s to be removed in its GitHub Actions',
+        repo,
+        `${NODE_VERSIONS_TO_REMOVE}`
+    );
+
+    return content;
+  }
+
+  /** @type {Exclude<import('@octokit/openapi-types').components["schemas"]["content-tree"]['entries'], undefined> | [import('@octokit/openapi-types').components["schemas"]["content-tree"]]} */
+  let files;
+
+  // Get all files from .github/workflows folder
+  try {
+    /** @type {import('@octokit/types').OctokitResponse<import('@octokit/openapi-types').components["schemas"]["content-tree"]>} */
+    // @ts-ignore Overriding the type of the response for the correct type with the `object` media type
+    const { data } = await octokit.request(
+        'GET /repos/{owner}/{repo}/contents/{path}',
+        {
+          owner,
+          repo,
+          path: PATH,
+          mediaType: {
+            format: 'object'
+          }
+        }
+    );
+
+    // We know that the path is a directory, we do this check to appease typescript and get rid of the `undefined`
+    if (Array.isArray(data.entries)) {
+      files = data.entries.filter(entry => isYamlFile(entry.name));
+    } else {
+      files = [data];
+    }
```

Comment:
From what I see in `PATH` variable, it is a file and not a directory, right?
Should we check all the `yaml` files under `.github/workflows` just in case?

Reply:
>From what I see in PATH variable, it is a file and not a directory, right?
> Should we check all the yaml files under .github/workflows just in case?

That might actually be why i'm getting the 404's, I pointed it to a file and not the directory